src/test/unit/engine/HelloRegionTest.cpp

```diff
@@ -36,57 +38,68 @@ namespace testing {
+    Directory::create("TestOutputDir");
+  std::string path = Path::join("TestOutputDir", "Data.csv");
+
+  // enter some arbitrary test data.
+  std::vector<std::vector<Real32>> testdata = {{0, 1.5f, 2.5f},
+                                               {1, 1.5f, 2.5f},
+                                               {2, 1.5f, 2.5f},
+                                               {3, 1.5f, 2.5f}};
+  size_t data_rows = testdata.size();
+  size_t data_cols = testdata[0].size();
+  std::ofstream f(path);
+  for (auto row : testdata) {
+    for (auto ele : row) {
+      f << " " << ele;
+    }
+    f << std::endl;
+  }
+  f.close();
+
+
   // Create network
   Network net;
 
+
+  std::string params = "{activeOutputCount: "+std::to_string(data_cols)+"}";
+
   // Add VectorFileSensor region to network
   std::shared_ptr<Region> region =
-      net.addRegion("region", "VectorFileSensor", "{activeOutputCount: 1}");
+      net.addRegion("region", "VectorFileSensor", params);
 
   // Set region dimensions
   Dimensions dims;
-  dims.push_back(1);
-
+  dims.push_back(1);  // 1 means variable size, in this case set by activeOutputCount.
   std::cout << "Setting region dimensions" << dims.toString() << std::endl;
-
   region->setDimensions(dims);
 
-  // Load data
-  const size_t DATA_SIZE = 4; //Data is only 4 rows
-  // try to find data file in the source
-  std::string path = Path::join(Path::getParent(__FILE__), "Data.csv");
-  if (!Path::exists(path)) {
-    // look for it where the install may have put it.
-    path = Path::makeAbsolute("../data/Data.csv");
-  }
-  std::cout << "Loading data from " << path << std::endl;
 
+  // Load data
+  // This will read in all data into a vector of vectors.
   std::vector<std::string> loadFileArgs;
   loadFileArgs.push_back("loadFile");
   loadFileArgs.push_back(path);
-  loadFileArgs.push_back("2"); //2=file format
-
+  loadFileArgs.push_back("2"); //2=file format, space separated elements.
   region->executeCommand(loadFileArgs);
 
   // Initialize network
-  std::cout << "Initializing network" << std::endl;
-
   net.initialize();
 
-  ArrayRef outputArray = region->getOutputData("dataOut");
-
   // Compute
-  std::cout << "Compute" << std::endl;
-
-  region->compute();
+  region->compute();  // This should fetch the first row into buffer
 
   // Get output
-  const Real64 *buffer = (const Real64 *)outputArray.getBuffer();
-  for (size_t i = 0; i < outputArray.getCount(); i++) {
-//    EXPECT_FLOAT_EQ(buffer[0], 0); //TODO add test, which values should be here?
-    std::cout << "  " << i << "    " << buffer[i] << "" << std::endl;
-  }
-
+  ArrayRef outputArray = region->getOutputData("dataOut");
+  EXPECT_TRUE(outputArray.getType() == NTA_BasicType_Real32);
+  EXPECT_EQ(outputArray.getCount(), testdata[0].size());
+  const Real32 *buffer = (const Real32 *)outputArray.getBuffer();
+  for (size_t i = 0; i < outputArray.getCount(); i++)
+    EXPECT_FLOAT_EQ(buffer[i], testdata[0][i]);
```

Comment:
great to see this finally fixed. The Data.csv and "weird" values in this test have been a pain

Reply:
>  Data.csv

By not having a prepared file we don't have to run the tests from that directory.  So no requirement to CD to build/Release/bin.